composer/trainer/trainer.py

```diff
@@ -1348,7 +1579,7 @@ def _train_microbatches(self, microbatches: Sequence[Batch], ddp_sync: bool = Tr
 
             use_grad_scaling = self._use_grad_scaling(self.state.precision, self.state.scaler)
 
-            if not self.deepspeed_enabled:
+            if not self.state.is_model_deepspeed:
```

Comment:
having a trainer level argument on `deepspeed_enabled` was very helpful, can we keep that?

Reply:
Added this property back.

Reply:
should we be using it here instead of checking the model, like before, to demonstrate the recommended path to checking if deepspeed is being used?

Reply:
Done